addons/iap/tools/iap_tools.py

```diff
@@ -138,6 +109,9 @@ def iap_jsonrpc(url, method='call', params=None, timeout=15):
     Calls the provided JSON-RPC endpoint, unwraps the result and
     returns JSON-RPC errors as exceptions.
     """
+    if hasattr(threading.current_thread(), 'testing') and threading.current_thread().testing:
```

Comment:
@lobaudoux Why has this been done, because it is weird we need to re-patch this if in external tests (which are now active for l10n in the nightly runbot) we want to test the iap.  (and it seems you are patching below as well)

See https://github.com/odoo/enterprise/pull/46432/files

Reply:
Because we don't want to perform real RPC calls during testing.

There were few occurrences where some calls were made during the tests outside of the IAP scope, so it didn't make sense to patch them.

Reply:
I understand your use case, but it's the exception, in usual tests, we do want to disable external calls, it has caused a few issues in the past.

Maybe we could add an argument to `iap_jsonrpc` to disable the check so that you don't need the weird patch ?

Reply:
Personally, I think doing the weird patch is okay, and definitely makes more sense than adding an argument to `iap_jsonrpc` (since iap_jsonrpc isn't called directly in the test anyway). 

I would have preferred having a dedicated context manager like the old `enable_iap_test()`, rather than patching the entire thread, but it's not too bad IMO.

Reply:
You're right if the function isn't called directly, the new argument makes no sense.

I also think that the patch is okay. Since the test wants to perform the "real" thing, it makes sense to execute outside of the test context.

What's the `enable_iap_test()` context manager you're refering too ? It doesn't ring a bell.

Reply:
It's exactly what you just removed in this PR.. ^^
Which we made some months ago https://github.com/odoo-dev/odoo/commit/9177076caee4f83590704e3af8e51ec6f7eaa0bc

Reply:
Well, it was just a thought - @jco-odoo has already gone ahead and r-plussed the fix to the l10n_pe_edi external tests, so no pressure from us. 

Anyway, thanks so much for answering our questions :smile: